runware/types.py

```diff
@@ -142,147 +142,68 @@ class IRefiner:
+    startStep: Optional[int] = None
+    endStep: Optional[int] = None
+    startStepPercentage: Optional[int] = None
+    endStepPercentage: Optional[int] = None
+    controlMode: Optional[EControlMode] = None
+    guideImage: Optional[Union[str, File]] = None
+    guideImageUnprocessed: Optional[Union[str, File]] = None
+
+    def __post_init__(self):
+        if (self.guideImage is None and self.guideImageUnprocessed is None) or \
+           (self.guideImage is not None and self.guideImageUnprocessed is not None):
+            raise ValueError("Exactly one of 'guideImage' or 'guideImageUnprocessed' must be provided.")
+        if (self.startStep and self.startStepPercentage) or (self.endStep and self.endStepPercentage):
+            raise ValueError("Exactly one of 'startStep/endStep' or 'startStepPercentage/endStepPercentage' must be provided.")
 
 
 @dataclass
 class IControlNetA(IControlNetGeneral):
-    def __post_init__(self):
-        # This method will run after the default __init__ generated by @dataclass
-        guide_images = [self.guide_image, self.guide_image_unprocessed]
-        # Check that exactly one of the fields is provided
-        if sum(x is not None for x in guide_images) != 1:
-            raise ValueError(
-                "Exactly one of 'guide_image' or 'guide_image_unprocessed' must be provided, not both or none."
-            )
+    preprocessor: EPreProcessor
 
 
 @dataclass
-class IControlNetCanny:
-    weight: Optional[float]
-    start_step: Optional[int]
-    end_step: Optional[int]
-    control_mode: Optional[EControlMode]
+class IControlNetCanny(IControlNetGeneral):
     model: Union[int, str]
-    low_threshold_canny: Optional[int]
-    high_threshold_canny: Optional[int]
+    lowThresholdCanny: Optional[int] = None
+    highThresholdCanny: Optional[int] = None
     preprocessor: EPreProcessor = EPreProcessor.canny
-    guide_image: Optional[Union[str, File]] = None
-    guide_image_unprocessed: Optional[Union[str, File]] = None
-    include_hands_and_face_open_pose: bool = field(default=True)
-
-    def __post_init__(self):
-        # This method will run after the default __init__ generated by @dataclass
-        guide_images = [self.guide_image, self.guide_image_unprocessed]
-        # Check that exactly one of the fields is provided
-        if sum(x is not None for x in guide_images) != 1:
-            raise ValueError(
-                "Exactly one of 'guide_image' or 'guide_image_unprocessed' must be provided, not both or none."
-            )
 
 
 @dataclass
-class IControlNetHandsAndFace:
-    preprocessor: EOpenPosePreProcessor
-    include_hands_and_face_open_pose: bool
-    weight: Optional[float]
-    start_step: Optional[int]
-    end_step: Optional[int]
-    control_mode: Optional[EControlMode]
-    guide_image: Optional[Union[str, File]] = None
-    guide_image_unprocessed: Optional[Union[str, File]] = None
+class IControlNetHandsAndFace(IControlNetGeneral):
```

Comment:
this shouldn't be called IControlNetOpenPose ? hands and face are properties of openpose, just like `lowThresholdCanny` and `highThresholdCanny` are properties of `IControlNetCanny`